src/algebra/category/Module/epi_mono.lean

```diff
@@ -0,0 +1,34 @@
+/-
+Copyright (c) 2021 Scott Morrison All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.category.Module.adjunctions
+import category_theory.epi_mono
+
+/-!
+# Monomorphisms in `Module R`
+
+This file shows that an `R`-linear map is a monomorphism in the category of `R`-modules
+if and only if it is injective.
+-/
+
+universe u
+
+open category_theory
+open Module
+
+namespace Module
+
+variables {R : Type u} [ring R]
+
+lemma mono_iff_injective {X Y : Module.{u} R} (f : X ⟶ Y) : mono f ↔ function.injective f :=
```

Comment:
This should also work with two different universes, right? (Or does the current proof somehow break?)

Reply:
We can't speak of morphisms between modules in different universes using the category theory API!

Reply:
We have `linear_map.ker_eq_bot_of_cancel`. Probably it should be at least mentioned in the docstring. We also have `linear_map.range_eq_top_of_cancel` (can be used for `epi_iff_surjective`).

Reply:
Thanks @urkud. I've done `epi_iff_surjective` following your suggestion, although it is much less pleasant than I'd hoped.

Reply:
@semorrison Sorry, I should have been more explicit. What I meant is that `R` can be in a different universe from `X` and `Y`, as far as I understand. I think that Markus decoupled those universes so that we can have `X Y : Module.{v}` for `R : Type u`.
But maybe I misremembered.

Reply:
Ah, I'd forgotten that. Okay, `epi_iff_surjective` has now been generalized to allow different universes, and I've adjust the comment on `mono_iff_injective` to say that if one wants to generalize to allow different universes you can use `linear_map.ker_eq_bot_of_cancel`. I'm disinclined to do that here.

Reply:
(It doesn't work here because the adjunction has some universe level constraints.)